CMakeLists.txt

```diff
@@ -42,6 +43,13 @@ elseif(${CMAKE_Fortran_COMPILER_ID} MATCHES "PGI")
   add_definitions(-DMUSICA_USING_PGI)
 endif()
 
+# YAML flag
+if(ENABLE_YAML)
+  add_definitions(-DUSE_YAML)
+else()
+  add_definitions(-DUSE_JSON)
+endif()
+
```

Comment:
Can you move these below the definition of the `musica` library and then set this with `target_compile_definitions(musica PRIVATE USE_*)`?

Reply:
done!